traffic_ops/v2-client/profile.go

```diff
@@ -164,7 +164,8 @@ func (to *Session) GetProfileByName(name string) ([]tc.Profile, ReqInf, error) {
 
 // GetProfileByParameter GETs a Profile by the Profile "param".
 func (to *Session) GetProfileByParameter(param string) ([]tc.Profile, ReqInf, error) {
-	URI := fmt.Sprintf("%s?param=%s", API_PROFILES, url.QueryEscape(param))
+	paramId, _ := strconv.Atoi(url.QueryEscape(param))
```

Comment:
We should probably check for the error here.

Reply:
Also, as I suggested earlier, instead of breaking the way pre existing client methods work, why not create a new client method that takes in a param ID instead of a param name, and then make the `GET` call? Something like this:

```
func (to *Session) GetProfileByParameterID(paramID string, header http.Header) ([]tc.Profile, ReqInf, error) {
         uri := fmt.Sprintf("%s?param=%d", APIProfiles, paramID)
	var data tc.ProfilesResponse
	reqInf, err := to.get(uri, header, &data)
	return data.Response, reqInf, err
}
```

Reply:
I don't think a new client is required, and I verified all the existing call to this method are passed.

Reply:
I don't want to create a new method, since the bug fixes requires the existing method to be changed to accept int parameters, and I verified the existing methods call are updated as per new fix.

Reply:
If we keep the existing methods as per your advice, if someone accidentally calls the existing method, it will throw an error, since that API is not anymore supported string.

Reply:
Created a new client method